tracer/src/Datadog.Trace/Sampling/DefaultSamplingRule.cs

```diff
@@ -58,23 +61,34 @@ public float GetSamplingRate(Span span)
                 return sampleRate;
             }
 
-            Log.Debug("Could not establish sample rate for trace {TraceId}", span.TraceId);
+            Log.Debug("Could not establish sample rate for trace {TraceId}. Using default rate instead: {rate}", span.TraceId, _defaultSamplingRate);
 
-            return 1;
+            return _defaultSamplingRate;
         }
 
         public void SetDefaultSampleRates(IEnumerable<KeyValuePair<string, float>> sampleRates)
         {
             // to avoid locking if writers and readers can access the dictionary at the same time,
             // build the new dictionary first, then replace the old one
             var rates = new Dictionary<SampleRateKey, float>();
+            float defaultSamplingRate = 1;
```

Comment:
```suggestion
            float defaultSamplingRate = _defaultSamplingRate;
```

Reply:
Done in d51d78926c9d73da91039af9eade6c467047f64b